xwayland/xwm.c

```diff
@@ -306,6 +309,9 @@ static void xsurface_set_net_wm_state(struct wlr_xwayland_surface *xsurface) {
 	if (xsurface->maximized_horz) {
 		property[i++] = xwm->atoms[NET_WM_STATE_MAXIMIZED_HORZ];
 	}
+	if (xsurface->minimized) {
+		property[i++] = xwm->atoms[NET_WM_STATE_HIDDEN];
```

Comment:
The `property` array must be expanded to size `5` if the properties are not mutually exclusive, otherwise there will be buffer overflow.

Reply:
changed it